test/integration/scheduler/scheduler_test.go

```diff
@@ -305,6 +305,53 @@ var _ = ginkgo.Describe("Scheduler", func() {
 			})
 		})
 
+		ginkgo.It("Should admit workloads when resources are dynamically reclaimed", func() {
+			firstWl := testing.MakeWorkload("first-wl", ns.Name).Queue(prodQueue.Name).
+				PodSets(
+					*testing.MakePodSet("first", 1).Request(corev1.ResourceCPU, "3").Obj(),
+					*testing.MakePodSet("second", 1).Request(corev1.ResourceCPU, "1").Obj(),
+					*testing.MakePodSet("third", 1).Request(corev1.ResourceCPU, "1").Obj(),
+				).
+				Obj()
+			ginkgo.By("Creating first workload", func() {
+				gomega.Expect(k8sClient.Create(ctx, firstWl)).Should(gomega.Succeed())
+
+				util.ExpectWorkloadsToHaveQuotaReservation(ctx, k8sClient, prodClusterQ.Name, firstWl)
+				util.ExpectPendingWorkloadsMetric(prodClusterQ, 0, 0)
+				util.ExpectReservingActiveWorkloadsMetric(prodClusterQ, 1)
+			})
+
+			ginkgo.By("Reclaim one pod from the first workload", func() {
+				gomega.Expect(workload.UpdateReclaimablePods(ctx, k8sClient, firstWl, []kueue.ReclaimablePod{{Name: "second", Count: 1}})).To(gomega.Succeed())
+
+				util.ExpectPendingWorkloadsMetric(prodClusterQ, 0, 0)
+				util.ExpectReservingActiveWorkloadsMetric(prodClusterQ, 1)
+			})
+
+			secondWl := testing.MakeWorkload("second-wl", ns.Name).Queue(prodQueue.Name).
+				PodSets(
+					*testing.MakePodSet("first", 1).Request(corev1.ResourceCPU, "3").Obj(),
+					*testing.MakePodSet("second", 1).Request(corev1.ResourceCPU, "1").Obj(),
+					*testing.MakePodSet("third", 1).Request(corev1.ResourceCPU, "1").Obj(),
+				).
+				Priority(100).
+				Obj()
+			ginkgo.By("Creating the second workload", func() {
+				gomega.Expect(k8sClient.Create(ctx, secondWl)).Should(gomega.Succeed())
+
+				util.ExpectWorkloadsToBePending(ctx, k8sClient, secondWl)
```

Comment:
Let's also check that secondWl is admitted

Reply:
I should to figure out this moment, because if we check here the admission status it always be an error

Reply:
Yeah, I'm wondering if the test case really reflects the issue description `Create a workload B of size 3, with higher priority, which preempts workload A.`. Its not clear if the firstWl is preempted. I think you are using CQ without preemption by priority enabled, so I expect `secondWl` is pending and `firstWl` remains admitted. I think the scenario assumes CQ with preemption.

Reply:
thank you for clarification, updated the test in accordance with comments https://github.com/kubernetes-sigs/kueue/pull/1241/commits/6c9fefb9095d932b0e0c469b1638ee01bc9457a6